doc/spec_TransferElement.dox

```diff
@@ -138,48 +138,53 @@ This documnent is currently still **INCOMPLETE**!
-   - \anchor transferElement_B_9_4 9.4 Write operations will throw  a ChimeraTK::runtime error in doWriteTrasferYyy() until open() has been called successfully. [\ref UnifiedTest_TransferElement_B_9_4_single "U"]
-
-- \anchor transferElement_B_10 10. DeviceBackends
-  - \anchor transferElement_B_10_1 10.1 DeviceBackend::setException() triggers the actions described in \ref transferElement_B_9_2 "9.2", \ref transferElement_B_9_3 "9.3" and \ref transferElement_B_9_4 "9.4" [\ref UnifiedTest_TransferElement_B_10_1 "U" (with all sub-points)]
-    - \anchor transferElement_B_10_1_1 10.1.1 Meta-backends like the LogicalNameMappingBackend delegate this call to all of their target backends
-    - 10.1.2 Further calls to DeviceBackend::setException() have no effect until Device::open() has successfully been called.
-    - \anchor transferElement_B_10_1_3 10.1.3 Application code can call Device::setException() to trigger the exception state \ref transferElement_comment_B_10_1_3 "(*)"
-  - \anchor transferElement_B_10_2 10.2 TransferElements know which backend to report exceptions to
-    - \anchor transferElement_B_10_2_1 10.2.1 TransferElement::_exceptionBackend is set when the TransferElement is created by a backend by calling TransferElement::setExceptionBackend().
-    - 10.2.2 Meta-backends like the LogicalNameMappingBackend can replace the exceptionBackend so the exception reporting is re-directed to the meta-backend.
-    - 10.2.3 TransferElement::setExceptionBackend() is virtual. Decorator-like TransferElements overload it so all target elements also get the correct exception backend.
-    - \anchor transferElement_B_10_2_4 10.2.4 Decorators set their TransferElement::_exceptionBackend in the constructor to their target's _exceptionBackend. \ref transferElement_comment_B_10_2_4 "(*)"
-    - 10.2.5 After replacing a low level transfer element in TransferElement::replaceTransferElement(), the exception backend of the new transfer element must be adapted.
-  - 10.3 After DeviceBackend::setException() has been called, DeviceBackend::isFunctional() returns false until the backend has successfully been recovered by DeviceBackend::open().
+  - \anchor transferElement_B_9_1 9.1 The transfer element which detects the ChimeraTK::runtime_error reports it to its TransferElement::_exceptionBackend by calling DeviceBackend::setException() \ref transferElement_comment_B_9_1 "(*)" (see \ref transferElement_B_9_2 "9.2") [\ref UnifiedTest_TransferElement_B_9_1_syncRead "U", \ref UnifiedTest_TransferElement_B_9_1_asyncRead "U", \ref UnifiedTest_TransferElement_B_9_1_write "U"]
+    - \anchor transferElement_B_9_1_1 9.1.1 TransferElement::_exceptionBackend is set when the TransferElement is created by a backend by calling TransferElement::setExceptionBackend().
+    - \anchor transferElement_B_9_1_2 9.1.2 Meta-backends like the LogicalNameMappingBackend can replace the exceptionBackend so the exception reporting is re-directed to the meta-backend.
```

Comment:
It is optional as a "Feature", i.e. you can if you have to

Reply:
If the meta-backend only has one target and does not hand out its own accessors, but just places a decorator around the TE created by the target, **it does not have to**. All it would add is one more indirection in the setException call.

Reply:
Well, if you want to allow that exception for the sake of performance optimisation (IMO completely unnecessary in that context), it needs to be exactly defined. In that case it e.g. also needs to delegate isFunctional().

Reply:
performance does not matter. it is the exception path anyway.

Easier rules are better. For consistency the I will remove the **can**.

Reply:
I would not have mentions all the possible side tracks that **can** allows. But this leaves it mushy, I agree. Better have it solid, but a bit rigid.